includes/REST_API/Font_Controller.php

```diff
@@ -0,0 +1,470 @@
+<?php
+/**
+ * Class Font_Controller
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\REST_API;
+
+use WP_Error;
+use WP_Post;
+use WP_Query;
+use WP_REST_Posts_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * Font_Controller class.
+ */
+class Font_Controller extends WP_REST_Posts_Controller {
+	/**
+	 * Registers the routes for posts.
+	 *
+	 * @since 1.16.0
+	 *
+	 * @see register_rest_route()
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_items' ],
+					'permission_callback' => [ $this, 'get_items_permissions_check' ],
+					'args'                => $this->get_collection_params(),
+				],
+				[
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => [ $this, 'create_item' ],
+					'permission_callback' => [ $this, 'create_item_permissions_check' ],
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				],
+				'schema' => [ $this, 'get_public_item_schema' ],
+			]
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\d]+)',
+			[
+				'args'   => [
+					'id' => [
+						'description' => __( 'Unique identifier for the font.', 'web-stories' ),
+						'type'        => 'integer',
+					],
+				],
+				[
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => [ $this, 'delete_item' ],
+					'permission_callback' => [ $this, 'delete_item_permissions_check' ],
+				],
+				'schema' => [ $this, 'get_public_item_schema' ],
+			]
+		);
+	}
+
+	/**
+	 * Returns a list of Google fonts.
+	 *
+	 * @since 1.16.0
+	 *
+	 * @return array List of Google fonts.
+	 */
+	protected function get_builtin_fonts() {
+		$file = WEBSTORIES_PLUGIN_DIR_PATH . 'includes/data/fonts/fonts.json';
```

Comment:
Maybe we should convert this to a PHP file with an array for performance reasons.

Reply:
I thought about that too. I'll do some micro benchmarking to see how big of an impact that would make.

Reply:
Quick test measuring how long it takes to load the fonts list:

Loading and parsing a JSON file: avg. 0.005 seconds

From PHP file returning an array: avg 0.00005 seconds

While the PHP file is a tad faster, it doesn't seem worth changing this at the expense of a more complicated build step.

Reply:
Tad faster, that seems a lot faster...

Reply:
0.005 is still fast enough. That's only 5 milliseconds.